backfill.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+# this script is used to update the article-json stored in lax
+# performing just an INGEST of content, *no* PUBLISH events are
+# sent whatsoever.
+
+# talking to lax via the adaptor.py script for many thousands of articles is
+# extremely slow, so this script bypasses all that, bulk generates articles,
+# bulk validates them then tells lax to do a bulk ingest.
+
+set -euo pipefail # strict mode
+#set -xv # debugging
+
+# housekeeping
+
+trap ctrl_c INT
+function ctrl_c() {
+    echo "caught ctrl-c"
+    exit 1
+}
+
+srcdir=$(pwd)
+errcho(){ >&2 echo $@; }
+
+# where articles will be linked to/downloaded for backfill
+defaultrunpath="run-$(date +'%Y%m%d%H%M%S')"
+runpath=${1:-$defaultrunpath}
+runpath=$(realpath $runpath) # realpath needs to be installed :(
+mkdir -p "$runpath"
+
+shift # download-api-raml.sh takes an arg :P
+
+# confirm
+
+echo "backfill.sh
+
+this script will:
+1. pull latest article-xml from elifesciences/elife-article-xml (large repo)
+2. download any missing/unpublished articles after consulting Lax (needs /srv/lax, s3 auth)
+3. create a 'backfill-run' directory with symbolic links to the xml to be processed
+4. generate article-json from ALL xml in the ./articles-xml/articles/ directory (long process)
+5. validate all generated article-json, failing if any are invalid
+6. force an INGEST into Lax for all valid articles (needs /srv/lax)"
+
+read -p "any key to continue (ctrl-c to quit) "
+
+# begin
+
+. download-elife-xml.sh
+
+# activate venv
+set +o nounset; . install.sh; set -o nounset; # virtualenv has unset vars we can't control
+
+# where to find xml on the fs
+xmlrepodir="$srcdir/article-xml/articles"
+
+# where to download unpublished xml to
+unpubxmldir="$srcdir/unpub-article-xml"
+mkdir -p "$unpubxmldir" # (create if necessary)
+
+# where generated article-json will be stored
+ajsondir="$runpath/ajson"
```

Comment:
currently the `article-json` folder is symlinked to `/ext` because of its size. Maybe `$runpath/ajson` could be created as `./article-json/run-2017.../` and symlinked in `$runpath` instead so that every new run ends up on the external volume. Especially since new runs won't overwrite older ones, so we need to keep space under control.